### ISSUE TITLE
Support for more input files

### ISSUE THREAD

**Issue by dolanor** [cassava/lackey#2]
Hi,

This is exactly a project I wanted to do for exactly the same reasons.
Most of my collection is made of .ogg files of good quality compression (because the flac way was still too much for my storage). And I also have some mp3 with some 320kbps encoding (yeah right…).
I'd like to reduce all of that to either lower quality ogg or low quality mp3 depending if the device will be an android or something else.

What is missing for that?
I think the mp3 is done for output, but not for input. Am I right? Also the same for Vorbis.

**Comment by cassava** [comment]
Hi dolanor,
So far it's working pretty good for me. I try to keep my collection composed of FLACs and MP3s. I output for a while to MP3s, but ffmpeg messes up converting some of the files, and at the bitrate I want to output, I found that the sound quality wasn't as good as I wanted. So now I output to Ogg, using the opus codec. But MP3 output is still working.

- MP3 should work for input, not 100% sure about OGG though. It should work though.
- I haven't tested any of this on Windows, as I only use it on Linux.
- If you want to go extra safe, make a read-only bind mountpoint of your music folder, as use that as the library input.

Here's the script I use:
```bash
#!/bin/sh

musicsource=/home/media/music

# Make source library read-only so I know I'm not f*cking up somehow...
musicsource_tmp=${musicsource}_readonly
if [ -d ${musicsource_tmp} ]; then
    echo "Something wrong here!"
    exit 1
else
    sudo mkdir ${musicsource_tmp}
    sudo mount -o bind,ro ${musicsource} ${musicsource_tmp}
    if [ $? -ne 0 ]; then
        echo "Not able to mount read-only music directory..."
        exit 1
    fi
fi

#lackey -L ${musicsource_tmp} sync -d /home/media/music2go
lackey -L ${musicsource_tmp} sync -d -f --opus /home/media/music4less

sudo umount ${musicsource_tmp}
sudo rmdir ${musicsource_tmp}
```

**Comment by dolanor** [comment]
Hi,

I tried your command (yeah I know, 3 years later…). And all I got is ignoring all the file in output and getting nothing.

```
dolanor@eastpeak:~$ lackey -L /tmp/musicsrc/Forró/ sync -d -f /tmp/music2go/
Reading source library (this might take a while)... 
Reading destination library (this might take a while)... 
ignoring: Bicho de Pé/2008 - Que Seja/01 - Bicho de Pé - Que Seja.mp3
…

dolanor@eastpeak:~$ file /tmp/musicsrc/Forró/Bicho\ de\ Pé/2008\ -\ Que\ Seja/01\ -\ Bicho\ de\ Pé\ -\ Que\ Seja.ogg 
/tmp/musicsrc/Forró/Bicho de Pé/2008 - Que Seja/01 - Bicho de Pé - Que Seja.ogg: Ogg data, Vorbis audio, stereo, 44100 Hz, ~320000 bps, created by: Xiph.Org libVorbis I
```

**Comment by cassava** [comment]
Hi @dolanor,
I found the issue (it's been a while since I last touched the code so it took a while to find the place).
The default planner only supported converting from a certain very supported subset of IN -> OUT codecs.
I added Ogg files to the list of supported INPUT files, but this causes the encoder to fallback to ffmpeg, instead of a manual process. My experience at least in the past has been that ffmpeg sometimes fails encoding files, and leaves a zero byte file in the destination instead. You might want to check your output directory to see if this is the case:
```
find -size 0 -name "*.mp3"
```
If you try the master branch it should accept OGG input files. Let me know if this works for you!
